### PR TITLE
feat: added management of the mssql defender

### PR DESCRIPTION
**Pull request by ivasm-ko**

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -104,3 +104,13 @@ variable "key_vault_id" {
   description = "Key Vault ID"
   default     = {}
 }
+
+variable "mssql_defender_state" {
+  description = "Manages Microsoft Defender state on the mssql server"
+  type        = string
+  default     = "Disabled"
```
Comment (owlleg6) [inline variables.tf:111]: set it to null by default, we will use "Disabled" within our infrastructure

Comment (ivasm-ko) [reply variables.tf:111]: done

File: variables.tf [hunk 2]
```diff
@@ -104,3 +104,13 @@ variable "key_vault_id" {
   description = "Key Vault ID"
   default     = {}
 }
+
+variable "mssql_defender_state" {
+  description = "Manages Microsoft Defender state on the mssql server"
+  type        = string
+  default     = "Disabled"
+  validation {
```
[the remark this answers is not in the document]
Comment (ivasm-ko) [reply variables.tf:112]: done

File: variables.tf [hunk 3]
```diff
@@ -104,3 +104,13 @@ variable "key_vault_id" {
...
   default     = {}
 }
+
+variable "mssql_defender_state" {
+  description = "Manages Microsoft Defender state on the mssql server"
+  type        = string
+  default     = "Disabled"
+  validation {
+    condition     = var.mssql_defender_state != null ? contains(["Enabled", "Disabled"], var.mssql_defender_state) : true
+    error_message = "Defender state on the mssql server can only be Disabled or Enabled"
```
[the remark this answers is not in the document]
Comment (ivasm-ko) [reply variables.tf:114]: done

**Comment by owlleg6** [comment]
LGTM

**Review by owlleg6 — APPROVED** [review]
LGTM

**Pull request merged**